js/src/components/contact-information/phone-number-card.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * External dependencies
+ */
+import { getCountryCallingCode } from 'libphonenumber-js';
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import { Flex, FlexItem, FlexBlock, CardDivider } from '@wordpress/components';
+import { Spinner } from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import useCountryCallingCodes from '.~/hooks/useCountryCallingCodes';
+import useGoogleMCPhoneNumber from '.~/hooks/useGoogleMCPhoneNumber';
+import Section from '.~/wcdl/section';
+import SelectControl from '.~/wcdl/select-control';
+import AppInputControl from '.~/components/app-input-control';
+import AccountCard, { APPEARANCE } from '.~/components/account-card';
+import AppButton from '.~/components/app-button';
+import AppSpinner from '.~/components/app-spinner';
+import './phone-number-card.scss';
+
+const noop = () => {};
+
+function PhoneNumberContent( {
+	initCountry,
+	initNationalNumber,
+	onPhoneNumberChange,
+} ) {
+	const countryCallingCodes = useCountryCallingCodes( 'select-options' );
+	const [ country, setCountry ] = useState( initCountry );
+	const [ number, setNumber ] = useState( initNationalNumber );
+
+	const handleChange = ( nextCountry, nextNumber ) => {
+		if ( nextCountry !== country ) {
+			setCountry( nextCountry );
+		}
+		if ( nextNumber !== number ) {
+			setNumber( nextNumber );
+		}
+
+		const countryCallingCode = nextCountry
+			? getCountryCallingCode( nextCountry )
+			: '';
+		onPhoneNumberChange( countryCallingCode, nextNumber, nextCountry );
+	};
+
+	const handleCountryChange = ( v ) => handleChange( v, number );
+	const handleNumberChange = ( v ) => handleChange( country, v );
+
+	return (
+		<Section.Card.Body>
+			<Flex gap={ 4 }>
+				<FlexItem>
+					<SelectControl
+						label={ __(
+							'Country code',
+							'google-listings-and-ads'
+						) }
+						isSearchable
+						excludeSelectedOptions={ false }
+						options={ countryCallingCodes }
+						selected={ country }
+						onChange={ handleCountryChange }
+					/>
+				</FlexItem>
+				<FlexBlock>
+					<AppInputControl
+						label={ __(
+							'Phone number',
+							'google-listings-and-ads'
+						) }
+						value={ number }
+						onChange={ handleNumberChange }
+					/>
+				</FlexBlock>
+			</Flex>
+		</Section.Card.Body>
+	);
+}
+
+export default function PhoneNumberCard( {
+	isPreview,
+	initEditing,
+	onEditClick,
+	onPhoneNumberChange = noop,
+} ) {
+	const [ isEditing, setEditing ] = useState( initEditing );
+	const { loaded, data } = useGoogleMCPhoneNumber();
```

Comment:
:nail_care: Having it here and in a containing element, https://github.com/woocommerce/google-listings-and-ads/pull/917/files#diff-51ff763be0b20871d193755311c5c437caafd6c4d46bb1413de599e1124a7714R25 seems redundant.

Why don't we make this a dumb component that gets fed with data and calls `*Change` callback?

Reply:
Changed by https://github.com/woocommerce/google-listings-and-ads/pull/917/commits/2a5ab61de8a0433ecea71218abc2fce914b3ce15